packages/solana-intents/src/lib.rs

```diff
@@ -176,3 +177,55 @@ impl<P> From<borsh::io::Error> for IntentError<P> {
         IntentError::DeserializeFailedError(err)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_offchain_message_roundtrip() {
+        let offchain_message = OffchainMessage::new(0, "Fogo Sessions".as_bytes()).unwrap();
+
+        assert_eq!(
+            Message::deserialize(&offchain_message.serialize().unwrap()).unwrap(),
+            Message::OffchainMessage(offchain_message.clone())
+        );
+
+        let mut message_serialized_short: Vec<u8> = offchain_message.serialize().unwrap();
+        message_serialized_short.pop();
+        assert_eq!(
+            Message::deserialize(&message_serialized_short)
+                .unwrap_err()
+                .kind(),
+            std::io::ErrorKind::InvalidData
+        );
+
+        let message_serialized_long: Vec<u8> = offchain_message
+            .serialize()
+            .unwrap()
+            .into_iter()
+            .chain(std::iter::once(0u8))
+            .collect();
+        assert_eq!(
+            Message::deserialize(&message_serialized_long)
+                .unwrap_err()
+                .kind(),
+            std::io::ErrorKind::InvalidData
+        );
+    }
+
+    #[test]
+    fn test_raw_message() {
+        let message = b"Fogo Sessions";
+        assert_eq!(
+            Message::deserialize(message).unwrap(),
+            Message::Raw(message.to_vec())
+        );
+
+        let message_long = b"Fogo Sessions Fogo Sessions Fogo Sessions";
```

Comment:
what's the point of the long messages? just another test case? seems unnecessary